### PR TITLE
fixed mcafee esm fetch alarms

### PR DESCRIPTION
**Pull request by anara123**
## Status
In Progress

## Related Issues
https://github.com/demisto/etc/issues/15030

## Description
- fetch incidents will fetch alarms according to alarm triggeredDate
- added parameter to the integration - timezone - query to ESM must be according to ESM timezone
- removed unnecessary login() and logout calls
- clean some code
- parse_date_range function will support timezone argument

## Screenshots
Paste here any images that will help the reviewer
image

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] McAfee ESM-v10

### CODE REVIEW

**Comment by anara123** [comment]
still in progress, need to add tests

File: Integrations/integration-ESM-v10.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@ display: McAfee ESM-v10
...
 description: Run queries and receive alarms from Intel Security ESM
+detaileddescription: If timezone parameter should be set according to the timezone
```
Comment (yaakovi) [inline Integrations/integration-ESM-v10.yml]: the `if` should be removed here

File: Integrations/integration-ESM-v10.yml [hunk 2]
```diff
@@ -858,206 +915,232 @@ script:
...
+                        'severity': severity_to_level(alarm['severity']),
+                        'rawJSON': json.dumps(alarm)
+                    })
 
-        if mode in ('alarms', 'both'):
-            end = datetime.datetime.now().isoformat()
-            last_run['value'] = end
-            last_run['alarms'] = end
+            if mode in ('cases', 'both'):
+                # esm.login()
```
Comment (yaakovi) [inline Integrations/integration-ESM-v10.yml]: remove commented code

File: Scripts/script-CommonServerPython.yml [hunk 3]
```diff
@@ -1339,8 +1342,11 @@ script: |-
       if not range_split[1] in ['minute', 'minutes', 'hour', 'hours', 'day', 'days', 'month', 'months', 'year', 'years']:
           return_error('The unit of date_range is invalid. Must be minutes, hours, days, months or years')
 
-      end_time = datetime.now()
-      start_time = datetime.now()
+      if not isinstance(timezone, int) or not isinstance(timezone, float):
```
Comment (yaakovi) [inline Scripts/script-CommonServerPython.yml]:
```suggestion
      if not isinstance(timezone, (int, float)):
```

File: Integrations/integration-ESM-v10.yml [hunk 4]
```diff
@@ -349,141 +374,143 @@ script:
...
 
-
-        def update_alarms_status(self, alarm_ids):
-            cmd = 'alarm%(action)sTriggeredAlarm' % (action, )
-            query = json.dumps({'triggeredIds' : [{'value' : int(i)} for i in alarm_ids]})
-
-            self.login()
+        @logger
+        def update_alarms_status(self, action, alarm_ids):
+            pass
```
Comment (orenzohar) [inline Integrations/integration-ESM-v10.yml]: remove

**Pull request merged**